### ISSUE TITLE
Case where :container_class and :container_extra_class are set is not handled

### ISSUE THREAD

**Issue by mindok** [maxmarcon/live_select#2]
If you are using `style: :none` option and set both the `:container_class` and `:container_extra_class` (or any combination of _class and _extra_class) you get a match error at `LiveSelect.Component.class/4`

The following addition to the function resolves the error. I may be misunderstanding the usage, but happy to submit a PR if this looks right...

```elixir
  defp class(_style, _element, class_override, class_extend) do
    class_override <> " #{class_extend}"
  end
```

**Comment by mindok** [comment]
Ah - read the docs!

**Comment by maxmarcon** [comment]
Thanks @mindok ! This is intended: for each element you can use either the `_class` options to completely override the default class (determined by `style`) or you can use the `_extra_class` to extend the default class. Using both doesn't make sense (you would then only use `_class`). I updated the docs to make it clear and I extended the `class` function to raise an exception explaining the issue if the user specifies both.

If you're using `style: :none` there are of course no default classes to extend so you should always use `_class`. I made sure an exception is raised also if `style: :none` is used with `_extra_class`.

Let me know if this makes sense to you and thanks for spotting this :)

**Comment by maxmarcon** [comment]
> Ah - read the docs!

Actually, no: write better docs :) 

I just updated the docs as a response to your comment. Thanks for pointing this out.

As you probably realized, this is intended: for each element you can use either the _class options to completely override the default class (determined by style) or you can use the _extra_class to extend the default class. Using both doesn't make sense (you would then only use _class). I updated the docs to make it clear and I extended the class function to raise an exception explaining the issue if the user specifies both.

If you're using style: :none there are of course no default classes to extend so you should always use _class. I made sure an exception is raised also if style: :none is used with _extra_class.

Let me know if this makes sense to you and thanks again for spotting this :)

Btw: I'm working on adding default styles that don't require daisyui - so there's soon going to be a `style: :tailwind` option that enables some default styles using vanilla tailwind classes. My plan right now is to basically mimic what the daisyui classes do. Essentially making the dropdown position absolute so it doesn't influence the flow of the page and little more.

**Comment by maxmarcon** [comment]
closed by 0c9ac85fef4b38c40482eb874ed3b4985969bbee
